src/Config/Converters/EntitySourceConverterFactory.cs

```diff
@@ -57,75 +57,62 @@ public EntitySourceConverter(bool replaceEnvVar)
+                    string? defaultValue = value?.ToString();
+                    paramList.Add(new ParameterMetadata
+                    {
+                        Name = prop.Name,
+                        Default = defaultValue,
+                    });
+                }
+
+                // Remove "parameters" from the JSON before deserialization
+                Dictionary<string, object> modObj = [];
+                foreach (JsonProperty property in root.EnumerateObject())
+                {
+                    if (!property.NameEquals("parameters"))
+                    {
+                        modObj[property.Name] = property.Value.Deserialize<object>(innerOptions) ?? new object();
+                    }
+                }
+
+                modObj["parameters"] = paramList;
+
+                string modJson = JsonSerializer.Serialize(modObj, innerOptions);
+
+                // Deserialize to EntitySource without parameters
+                EntitySource? entitySource = JsonSerializer.Deserialize<EntitySource>(modJson, innerOptions)
+                    ?? throw new JsonException("Failed to deserialize EntitySource from modified JSON.");
+
+                // Use the with expression to set the correct Parameters
+                return entitySource with { Parameters = paramList };
             }
-
-            return source;
-        }
-
-        private static object GetClrValue(JsonElement element)
-        {
-            return element.ValueKind switch
-            {
-                JsonValueKind.String => element.GetString() ?? string.Empty,
-                JsonValueKind.Number => GetNumberValue(element),
-                JsonValueKind.True => true,
-                JsonValueKind.False => false,
-                _ => element.ToString()
-            };
-        }
-
-        /// <summary>
-        /// Attempts to get the correct numeric value from the <see cref="JsonElement"/>.
-        /// If all possible numeric values are exhausted, the raw text is returned.
-        /// </summary>
-        /// <param name="element">JSON element to extract the value from.</param>
-        /// <returns>The parsed value as a CLR type.</returns>
-        private static object GetNumberValue(JsonElement element)
-        {
-            if (element.TryGetInt32(out int intValue))
-            {
-                return intValue;
-            }
-
-            if (element.TryGetDecimal(out decimal decimalValue))
-            {
-                return decimalValue;
-            }
-
-            if (element.TryGetDouble(out double doubleValue))
+            else
             {
-                return doubleValue;
+                string rawJson = root.GetRawText();
+                // If already in new format, deserialize as usual
+                EntitySource? source = JsonSerializer.Deserialize<EntitySource>(rawJson, innerOptions);
+
+                if (source?.Parameters is not null)
```

Comment:
Wouldnt deserialization of rawJson here fill up the source.Parameters with the correct list? What is the purpose of this `with` expression on line 110?